src/SprykerEco/Zed/PunchoutCatalogs/Communication/Form/PunchoutCatalogConnectionFormDataProvider.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerEco\Zed\PunchoutCatalogs\Communication\Form;
+
+use Generated\Shared\Transfer\CompanyBusinessUnitCriteriaFilterTransfer;
+use Generated\Shared\Transfer\CompanyBusinessUnitTransfer;
+use Generated\Shared\Transfer\PunchoutCatalogConnectionTransfer;
+use SprykerEco\Zed\PunchoutCatalogs\Dependency\Facade\PunchoutCatalogsToCompanyBusinessUnitFacadeInterface;
+
+class PunchoutCatalogConnectionFormDataProvider
+{
+    /**
+     * @var \SprykerEco\Zed\PunchoutCatalogs\Dependency\Facade\PunchoutCatalogsToCompanyBusinessUnitFacadeInterface
+     */
+    protected $companyBusinessUnitFacade;
+
+    /**
+     * @param \SprykerEco\Zed\PunchoutCatalogs\Dependency\Facade\PunchoutCatalogsToCompanyBusinessUnitFacadeInterface $companyBusinessUnitFacade
+     */
+    public function __construct(PunchoutCatalogsToCompanyBusinessUnitFacadeInterface $companyBusinessUnitFacade)
+    {
+        $this->companyBusinessUnitFacade = $companyBusinessUnitFacade;
+    }
+
+    /**
+     * @return array
+     */
+    public function getOptions(): array
+    {
+        return [
+            'data_class' => PunchoutCatalogConnectionTransfer::class,
+            PunchoutCatalogConnectionForm::OPTION_BUSINESS_UNIT_CHOICES => $this->prepareCompanyBusinessUnitChoices(),
+        ];
+    }
+
+    /**
+     * @return \Generated\Shared\Transfer\PunchoutCatalogConnectionTransfer
+     */
+    public function getInitialData(): PunchoutCatalogConnectionTransfer
```

Comment:
All FormDataProviders have only two methods `getOptions` and `getData`. Why it doesnot work for you?